Add Particles to Blocks with Enchanting Power

PepperCode1 opened the pull request:
This uses a single method client side mixin to EnchantingTableBlock.

PepperCode1 committed:
Add Particles to Blocks with Enchanting Power

This uses a single method client side mixin to EnchantingTableBlock.

i509VCB commented on src/main/java/io/github/fablabsmc/fablabs/mixin/block/extensions/enchantment/EnchantingTableBlockMixin.java:
```diff
@@ -0,0 +1,40 @@
+class EnchantingTableBlockMixin {
+	@Redirect(method = "randomDisplayTick(Lnet/minecraft/block/BlockState;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Ljava/util/Random;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/BlockState;isOf(Lnet/minecraft/block/Block;)Z"))
+	private boolean spawnParticle(BlockState state, Block block) {
+		if (state.getBlock() instanceof EnchantmentTablePowerExtension) {
```
I believe we should also be checking if the power level is at least 1 before trying to return true here.

You can get the world param easily by just appending the signature of `randomDisplayTick` on the end of the method params

PepperCode1 replied:
The check has been added in the latest version.

PepperCode1 committed:
Update EnchantingTableBlockMixin.java

Made EnchaningTableBlockMixin extend EnchantingTableBlock's superclass, BlockWithEntity.
Added check to see if the block's enchantment power is greater than 0.
